include/rmm/mr/device/detail/free_list.hpp

```diff
@@ -55,34 +70,41 @@ struct block {
    * @return true Returns true if this blocks's `ptr` + `size` == `b.ptr`, and `not b.is_head`,
                   false otherwise.
    */
-  bool is_contiguous_before(block const& b) const noexcept
+  inline bool is_contiguous_before(block const& b) const noexcept
   {
     return (this->ptr + this->size == b.ptr) and not(b.is_head);
   }
 
+  /**
+   * @brief Is this block large enough to fit `sz` bytes?
+   *
+   * @param sz The size in bytes to check for fit.
+   * @return true if this block is at least `sz` bytes
+   */
+  inline bool fits(size_t sz) const noexcept { return size >= sz; }
+
+  /**
+   * @brief Is this block a better fit for `sz` bytes than block `b`?
+   *
+   * @param sz The size in bytes to check for best fit.
+   * @param b The other block to check for fit.
+   * @return true If this block is a tighter fit for `sz` bytes than block `b`.
+   * @return false If this block does not fit `sz` bytes or `b` is a tighter fit.
+   */
+  inline bool is_better_fit(size_t sz, block const& b) const noexcept
+  {
+    return fits(sz) && (size < b.size || b.size < sz);
+  }
+
   /**
    * @brief Print this block. For debugging.
    */
   void print() const { std::cout << reinterpret_cast<void*>(ptr) << " " << size << "B\n"; }
-};
 
-/**
- * @brief Coalesce two contiguous blocks into one.
- *
- * `a` must immediately precede `b` and both `a` and `b` must be from the same upstream allocation.
- * That is, `a.ptr + a.size == b.ptr` and `not b.is_head`. Otherwise behavior is undefined.
- *
- * @param a first block to merge
- * @param b second block to merge
- * @return block The merged block
- */
-inline block merge_blocks(block const& a, block const& b)
-{
-  assert(a.ptr + a.size == b.ptr);  // non-contiguous blocks
-  assert(not b.is_head);            // Can't merge across upstream allocation boundaries
-
-  return block{a.ptr, a.size + b.size};
-}
+  char* ptr;     ///< Raw memory pointer
```

Comment:
Since the default constructor is being set to `=default`, and these members aren't initialized, a default constructed `block` is in an invalid, unintialized state. So just initialize the members to avoid this. 
```suggestion
  char* ptr{};     ///< Raw memory pointer
```